src/components/posts/PostsPage.js

```diff
@@ -1,33 +1,46 @@
+
 import React from 'react';
 import { useState, useEffect } from 'react';
 import PostForm from './PostForm';
 import client from '../../utils/client';
 import './style.css';
-import { Box, Stack } from '@mui/material';
-import dateTimetoRelativeTime from './helperfunctions';
-import { Link } from 'react-router-dom';
+import Post from './Post';
 
 const PostsPage = ({ role }) => {
   const [post, setPost] = useState({ content: '' });
   const [postResponse, setPostResponse] = useState('');
   const [posts, setPosts] = useState([]);
-
+  console.log(posts);
   useEffect(() => {
-    client.get('/posts').then((res) => setPosts(res.data.data.posts));
-  }, []);
+    client.get('/posts').then((res) => {
+      setPosts(res.data.data.posts);
+    });
+  }, [postResponse]);
 
-  const createPost = async (event) => {
+  const createPost = (event) => {
     event.preventDefault();
     client
       .post('/post', post)
       .then((res) => {
         setPostResponse(res.data);
         setPosts((posts) => [res.data.data.post, ...posts]);
       })
-      .catch((data) => {});
+      .catch((data) => {
+      console.log(data);
```

Comment:
Could do with adding more context to this error log so it's easier to know what has happened in the console.

Reply:
I just replaced it with an error.message + 'Invalid post'